src/SignalR/clients/ts/signalr/src/HttpConnection.ts

```diff
@@ -401,32 +409,32 @@ export class HttpConnection implements IConnection {
         }
     }
 
-    private resolveTransport(endpoint: IAvailableTransport, requestedTransport: HttpTransportType | undefined, requestedTransferFormat: TransferFormat): HttpTransportType | null {
+    private tryResolveTransport(endpoint: IAvailableTransport, requestedTransport: HttpTransportType | undefined, requestedTransferFormat: TransferFormat): HttpTransportType | Error {
```

Comment:
It feels weird to have a "try"  prefixed method not return a bool.
Imo you should always be able to do 

```
if (tryFoo()) {
    // do something if Foo succeeded
}
else { 
    // do something if foo failed
}
```